include/openthread/platform/crypto.h

```diff
@@ -610,6 +615,20 @@ otError otPlatCryptoEcdsaGenerateKey(otPlatCryptoEcdsaKeyPair *aKeyPair);
  */
 otError otPlatCryptoEcdsaGetPublicKey(const otPlatCryptoEcdsaKeyPair *aKeyPair, otPlatCryptoEcdsaPublicKey *aPublicKey);
 
+/**
+ * Import the provided key-pair into PSA.
+ *
+ * @param[in]  aKeyRef            Key Ref to use while importing ECDSA key-pair.
+ * @param[in]  aDerBytes          A pointer to an ECDSA key-pair in DER format.
+ * @param[in]  aDerLength         Length of the key-pair to be stored.
+ *
+ * @retval OT_ERROR_NONE          Public key was imported successfully.
+ * @retval OT_ERROR_FAILED        The key-pair import failed.
+ * @retval OT_ERROR_INVALID_ARGS  The @p aDerBytes is NULL.
+ *
+ */
+otError otPlatCryptoImportKeyPair(uint32_t aKeyRef, const uint8_t *aDerBytes, uint8_t aDerLength);
```

Comment:
Can we extend the existing `otPlatCryptoImportKey()` to also handle importing of ECDSA key (in DER format) (instead of adding new platform API?)

```c++
otError otPlatCryptoImportKey(otCryptoKeyRef      *aKeyRef,
                              otCryptoKeyType      aKeyType,
                              otCryptoKeyAlgorithm aKeyAlgorithm,
                              int                  aKeyUsage,
                              otCryptoKeyStorage   aKeyPersistence,
                              const uint8_t       *aKey,
                              size_t               aKeyLen);
```

We can extended `otCryptoKeyType` to include "ECDA Key Pair" as a type and also update `otCryptoKeyAlgorithm`?

This way we have one common platform function for importing keys into PSA stoarage?

Reply:
I agree, but the intention of this API was a bit different. I wanted a single API to import the keys for testing. Currently we get the pointer and copy the test der using a memcpy in test_ecdsa. I wanted a single interface for testing, as in real-world I see that generating keys would be secure than setting keys from DER. I dont see this used for any other purpose. I can easily call ImportKey to import the keypair here, but I see now that the original purpose of the API is not valid anymore. Let me re-work this a bit more.

Reply:
Removed the new API in the latest push.